### PR TITLE
Fix failures of react-addons-shallow-compare

### LINKED ISSUE #63: react-vis brings the latest version of React when no react-addons-shallow-compare is installed
Originally reported by @harishkraman

### PR DESCRIPTION
**Pull request by bulyonov**
- The major source of issues (incorrect version of React) was found and replaced (0.15 -> 15)
- react-addons-shallow-compare is now moved to `peerDepencencies`.

This PR fixes #63 and #70.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,15 +40,16 @@
     "eslint": "^1.10.3",
     "eslint-plugin-react": "^3.16.1",
     "node-sass": "^3.4.2",
-    "react": ">=0.14.0 <0.15.0",
-    "react-addons-test-utils": "^0.14.7",
-    "react-dom": ">=0.14.0 <0.15.0",
+    "react": "0.14.x - 15.x",
+    "react-addons-test-utils": ">=0.14.0",
+    "react-dom": "0.14.x - 15.x",
```
Comment (chrisirhc) [inline package.json]: Missing `react-addons-shallow-compare` here

File: package.json [hunk 2]
```diff
@@ -41,15 +40,16 @@
...
     "run-browser-babel": "^4.0.1",
     "tape": "^4.0.0",
     "uglify-js": "^2.6.1"
   },
   "peerDependencies": {
-    "react": ">=0.13.0"
+    "react": ">=0.13.0",
+    "react-addons-shallow-compare": ">=0.14.0"
```
Comment (chrisirhc) [inline package.json]: These two should be of the same version requirements

File: package.json [hunk 3]
```diff
@@ -41,15 +40,17 @@
...
+    "react-addons-shallow-compare": "0.14.x - 15.x",
+    "react-dom": "0.14.x - 15.x",
     "run-browser-babel": "^4.0.1",
     "tape": "^4.0.0",
     "uglify-js": "^2.6.1"
   },
   "peerDependencies": {
-    "react": ">=0.13.0"
+    "react": "0.14.x - 15.x",
+    "react-addons-shallow-compare": "0.14.x - 15.x"
```
Comment (chrisirhc) [inline package.json]: How about `>=0.14.0` here? We can watch out for incompatibilities in the future

Comment (bulyonov) [inline package.json]: @chrisirhc done

**Pull request merged**